numba_dpex/config.py

```diff
@@ -87,3 +87,19 @@ def __getattr__(name):
 # Flag to turn on the ConstantSizeStaticLocalMemoryPass in the kernel pipeline.
 # The pass is turned off by default.
 STATIC_LOCAL_MEM_PASS = _readenv("NUMBA_DPEX_STATIC_LOCAL_MEM_PASS", int, 0)
+
+DPEX_OPT = _readenv("NUMBA_DPEX_OPT", int, 2)
+
+if DPEX_OPT > 2:
```

Comment:
We should not put the check here. Rather I prefer it in the codegen function. The reason is someone can set `config.DPEX_OPT=3` somewhere else in the code without this warning getting triggered.